File: examples/openai-sdk/streaming.js
// Пример использования OpenAI SDK с прокси для Qwen AI в потоковом режиме
// Установка: npm install openai

import OpenAI from 'openai';

const openai = new OpenAI({
    baseURL: 'http://localhost:3264/api', 
    apiKey: 'dummy-key', 
});

async function streamFromQwen() {
    try {
        console.log('Отправка потокового запроса к Qwen AI...\n');


        const stream = await openai.chat.completions.create({
            messages: [
                { role: 'user', content: 'Напиши небольшую историю о космических путешествиях' }
            ],
            model: 'qwen-max-latest', 
            stream: true, 
        });

        console.log('Ответ от Qwen (потоковый режим):\n');

        for await (const chunk of stream) {
            const content = chunk.choices[0]?.delta?.content || '';
            process.stdout.write(content);
        }

        console.log('\n\nПотоковый ответ завершен.');

    } catch (error) {
        console.error('Ошибка при выполнении потокового запроса:', error);
    }
}

// Запуск
streamFromQwen(); 

File: examples/python-sdk/streaming.py
from openai import OpenAI

client = OpenAI(
    base_url="http://localhost:3264/api",
    api_key="free-qwen-api",
)

stream = client.chat.completions.create(
    model="qwen-max-latest",
    messages=[{"role": "user", "content": "Напиши мини-историю про робота."}],
    stream=True,
)

for chunk in stream:
    delta = chunk.choices[0].delta
    if delta and delta.content:
        print(delta.content, end="", flush=True)

print()
